### PR TITLE
Protocol: retention updates

### PR DESCRIPTION
**Pull request by mlsteele**
Protocol updates. The only change that has an effect is adding `return` to `push.go`

### CODE REVIEW

File: go/protocol/chat1/extras.go [hunk 1]
```diff
@@ -217,6 +217,35 @@ func (m MessageUnboxed) IsValid() bool {
 	return false
 }
 
+// Whether the message is all of:
+// 1. Valid
+// 2. Has a non-deleted body with a type matching the header
+//    (TLFNAME is an exception as it has no body)
+// 3. Supersededby == 0
+func (m MessageUnboxed) IsValidFull() bool {
```
Comment (mmaxim) [inline go/protocol/chat1/extras.go]: I think maybe this should get a better name, its hard to know what is different from the name alone. Also this probably fails `go vet` with the comment.

Comment (mlsteele) [reply go/protocol/chat1/extras.go]: Any ideas for the name? IsValidAndNotDeleted.. but that's kind of not true. IsValidAndWorthyOfDisplay.

Comment (mmaxim) [reply go/protocol/chat1/extras.go]: `IsValidAndDisplayable`?

Comment (mlsteele) [reply go/protocol/chat1/extras.go]: It's true for `DELETE` and `METADATA` so I feel like displayable is a little misleading

Comment (mlsteele) [reply go/protocol/chat1/extras.go]: Also `go vet` doesn't seem to have any problem with this for me

Comment (mmaxim) [reply go/protocol/chat1/extras.go]: We might block it in `protocol`.

Comment (mlsteele) [reply go/protocol/chat1/extras.go]: I'm not sure I understand. Is there a failure somewhere?

**Pull request merged**